custom/icds_reports/static/js/angular-services/baseControllers.service.js

```diff
@@ -1,15 +1,31 @@
-/* global d3 */
+/* global d3, moment */
 
 window.angular.module('icdsApp').factory('baseControllersService', function() {
     return {
-        BaseController: function($scope, $routeParams, $location, locationsService, userLocationId,
+        BaseController: function ($scope, $routeParams, $location, locationsService, userLocationId,
             storageService, haveAccessToAllLocations, haveAccessToFeatures) {
             var vm = this;
             if (Object.keys($location.search()).length === 0) {
                 $location.search(storageService.getKey('search'));
             } else {
                 storageService.setKey('search', $location.search());
             }
+
+            vm.showInfoMessage = function () {
```

Comment:
Could this function be extracted into a separate file and shared among the directives?